src/sentry/tasks/files.py

```diff
@@ -0,0 +1,21 @@
+from __future__ import absolute_import
+
+from sentry.tasks.base import instrumented_task
+from sentry.tasks.deletion import MAX_RETRIES
+
+
+@instrumented_task(
+    name='sentry.tasks.files.delete_file',
+    queue='files.delete',
+    default_retry_delay=60 * 5,
+    max_retries=MAX_RETRIES
+)
+def delete_file(path, checksum, **kwargs):
+    from sentry.models.file import get_storage, FileBlob
+    from sentry.app import locks
+    from sentry.utils.retries import TimedRetryPolicy
+
+    lock = locks.get('fileblob:upload:{}'.format(checksum), duration=60 * 10)
+    with TimedRetryPolicy(60)(lock.acquire):
+        if FileBlob.objects.filter(checksum=checksum).exists():
```

Comment:
Don't you want the inverse of this? We need to make sure the `FileBlob` doesn't exist.